File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        trueSum = n * (n + 1) / 2
        
        listSum = 0
        for i in range(len(nums)):
            listSum += nums[i]
        
        
        return (int)(trueSum - listSum)
        

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int xor1 = 0;
        int xor2 = 0;
        
        for(int i = 0; i < nums.length; i++) {
            xor2 = xor2 ^ nums[i]; // XOR of all elements in the array
            xor1 = xor1 ^ (i + 1);  // XOR of all numbers from 1 to n
        }
        return xor1 ^ xor2; // XOR of the above two will give the missing number
    }
}
